diagnostic.md

```diff
@@ -40,31 +63,36 @@ like to show all movies favorited by a profile on
 `http://localhost:3000/profiles/1`
 
   ```md
-    # < Your Response Here >
+  Serializers filter and reshape data for what the server can send out or receive... so in the below code we have a serializer for Profiles that shows us the different attributes of a profile and picks out the id of every movie associated with a profile and puts it into an array for us to see.
   ```
 
   ```rb
   class ProfileSerializer < ActiveModel::Serializer
+    attributes :id, :given_name, :surname, :email, :movies
+
+    def movies
+      object.movies.pluck(id)
+    end
   end
   ```
 
 1.  What would the command be to _scaffold_ out a **join table** for Favorites from
 the above `Movies` and `Profiles`.
 
   ```sh
-    # < Your Response Here >
+    bin/rails generate scaffold favorite profile:references movie:references date_added:date
   ```
 
 1.  What is `Dependent: Destroy` and where/why would we use it?
 
   ```md
-    # < Your Response Here >
+    I think it's when we delete a row on the join table (not 100% sure)? So in this case we'd use it if a user decided she didn't like a movie anymore and wanted to remove it from her Favorites.
```

Comment:
`dependent: :destroy` will cause the deletion of an object to also delete all associated objects.  For example, if a `User` model has `has_many :favorites, dependent: :destroy`, then deleting a particular user will also delete all rows in the `favorites` table where `user_id` refers to the `id` of the user being deleted.